### PR TITLE
swi-prolog 7.3.18 (devel)

### PR DESCRIPTION
**Pull request by honnix**
Besides the version, a fix for compiler optimization is done. Get
details here: SWI-Prolog/packages-semweb#22

### CODE REVIEW

File: swi-prolog.rb [hunk 1]
```diff
@@ -62,13 +62,16 @@ def install
     ENV.append "DISABLE_PKGS", "jpl" if build.without? "jpl"
     ENV.append "DISABLE_PKGS", "xpce" if build.without? "xpce"
 
+    # Default value Os causes segfault
+    ENV["HOMEBREW_OPTIMIZATION_LEVEL"] = "O2"
```
Comment (DomT4) [inline swi-prolog.rb]: Should be able to shorten this to `ENV.O2`? I'd be a bit surprised if `superenv` honoured formula overriding of the `HOMEBREW_OPTIMIZATION_LEVEL` variable.

Comment (honnix) [inline swi-prolog.rb]: So change `ENV["HOMEBREW_OPTIMIZATION_LEVEL"] = "O2"` to just `ENV.O2`?

I don't really know how this `ENV` thing works to be honest. This PR is a more like a hack than proper code. I can try it out and resubmit the PR. On the other hand, we are still investigating the root cause because modifying compiler flag in this case is anyway a workaround.

Comment (DomT4) [inline swi-prolog.rb]: I believe just `ENV.O2` is what you're looking for.

Comment (honnix) [inline swi-prolog.rb]: Alright. Thanks a lot! It works.

File: swi-prolog.rb [hunk 2]
```diff
@@ -62,13 +62,17 @@ def install
     ENV.append "DISABLE_PKGS", "jpl" if build.without? "jpl"
     ENV.append "DISABLE_PKGS", "xpce" if build.without? "xpce"
 
+    # Default value Os causes segfault in latest swi-prolog devel version
+    # This is the workaround
+    ENV.O2
```
Comment (dunn) [inline swi-prolog.rb]: Has this been reported upstream?

Comment (honnix) [inline swi-prolog.rb]: Which upstream? Homebrew? Not really. Still not sure it's a compiler problem or
some hidden bug in swi-prolog.

Best regards, Hongxin
On Sat, Mar 12, 2016 at 21:56, Alex Dunn <notifications@github.com> wrote:
In swi-prolog.rb
[https://github.com/Homebrew/homebrew-x11/pull/195#discussion_r55927519] :

> @@ -62,13 +62,17 @@ def install
>      ENV.append "DISABLE_PKGS", "jpl" if build.without? "jpl"
>      ENV.append "DISABLE_PKGS", "xpce" if build.without? "xpce"
>  
> +    # Default value Os causes segfault in latest swi-prolog devel version
> +    # This is the workaround
> +    ENV.O2


Has this been reported upstream?

—
Reply to this email directly or view it on GitHub
[https://github.com/Homebrew/homebrew-x11/pull/195/files#r55927519] .[https://github.com/notifications/beacon/AAJsrBcZhyCsemLDfYT4BZE0exosDU6Rks5psyiQgaJpZM4HufMI.gif]

Comment (dunn) [inline swi-prolog.rb]: Upstream to the swi-prolog developers.

Comment (honnix) [inline swi-prolog.rb]: Sure. I believe they are well aware of this and we are currently trying to
find the root cause.

Comment (dunn) [inline swi-prolog.rb]: Great, thank you!

**Comment by dunn** [comment]
Merged in c14365ed3d3774d701cf5609e87a8b32ae44df90, thanks!
